elastio-stream-kafka/consumer.py

```diff
@@ -34,39 +36,47 @@
     )
-topic_already_exists = False
+topic_previously_backed_up = False
 rps = [json.loads(rp) for rp in res.stdout.splitlines()]
 for rp in rps[0]:
     if rp['kind']['kind'] == 'Stream':
         rp_name = rp['asset_snaps'][0]['asset_id'].split(':')[-1]
-        if rp_name == args.topic_name:
-            try:
-                end_timestamp = rp['tags']['end_timestamp']
-                topic_already_exists = True
-                break
-            except KeyError:
+        try:
+            if rp_name == args.stream_name and rp['tags']['topic_name'] == args.topic_name:
+                topic_previously_backed_up = True
                 break
+        except KeyError:
+            break
 
-fmsg = True
-if topic_already_exists:
-    for msg in consumer:
-        if msg.timestamp > int(end_timestamp):
-            # print(msg)
-            if fmsg:
-                store_first_message(msg, fmsg=fmsg)
-                fmsg = False
-            print({"topic": msg.topic, "key": msg.key, "value": msg.value, "partition": msg.partition, "timestamp": msg.timestamp})
-    store_last_message(msg=msg)
+partition_count = len(consumer.partitions_for_topic(topic_name))
+topic_info_data['partition_count'] = partition_count
+if topic_previously_backed_up:
+    for partition in range(partition_count):
+        first_message = True
+        consumer.assign([TopicPartition(args.topic_name, partition),])
+        for msg in consumer:
+            if msg.offset > int(rp['tags'][f'partition_{str(partition)}_last_msg_offset']):
+                # print(msg)
+                if first_message:
+                    topic_info_data[f'partition_{str(partition)}_first_msg_offset'] = msg.offset
+                    first_message = False
+                print({"topic": msg.topic, "key": msg.key, "value": msg.value, "partition": msg.partition, "timestamp": msg.timestamp, "offset": msg.offset, "headers":msg.headers})
+            topic_info_data[f'partition_{str(partition)}_last_msg_offset'] = msg.offset
+    write_topic_info(data=topic_info_data)
 else:
-    for msg in consumer:
-        # print(msg)
-        if fmsg:
-            store_first_message(msg, fmsg=fmsg)
-            fmsg = False
-        print({"topic": msg.topic, "key": msg.key, "value": msg.value, "partition": msg.partition, "timestamp": msg.timestamp})
-    store_last_message(msg=msg)
+    for partition in range(partition_count):
+        first_message = True
+        consumer.assign([TopicPartition(args.topic_name, partition),])
+        for msg in consumer:
+            # print(msg)
+            if first_message:
+                topic_info_data[f'partition_{str(partition)}_first_msg_offset'] = msg.offset
+                first_message = False
+            print({"topic": msg.topic, "key": msg.key, "value": msg.value, "partition": msg.partition, "timestamp": msg.timestamp, "offset": msg.offset, "headers":msg.headers})
+        topic_info_data[f'partition_{str(partition)}_last_msg_offset'] = msg.offset
```

Comment:
... While here, since this line is not in a loop, looks like it will crash right here on `msg` being undefined.